### ISSUE TITLE
return model with precalculated nearest neighbor 

### ISSUE THREAD

**Issue by yuhanH** [jlmelville/uwot#63]
Hi, 
I noticed that `umap` cannot return a model with precalculated nearest neighbor input in `nn_method`.

I would propose that modifications of two parts will enable `umap` to return the model with precalculated nearest neighbor input.

part1:
https://github.com/jlmelville/uwot/blob/cbb78b5f75521b963138c96c2644eba6e4142720/R/uwot.R#L1656
        `scale_info = if (!is.null(X)) { attr_to_scale_info(X) } else { NULL },`

part2:
https://github.com/jlmelville/uwot/blob/cbb78b5f75521b963138c96c2644eba6e4142720/R/uwot.R#L1687
```       
    if(!is.null(X)){
            res$metric[[1]] <- list(ndim = length(res$nn_index$getItemsVector(0)))
          } else{
            res$metric[[1]] <- list()
          }
```
Happy to put a PR for this if people agree.

**Comment by jlmelville** [comment]
Hi @yuhanH, `uwot` doesn't return a model when using precalculated nearest neighbors, because you can't transform new points with just that information. If you think there is a use for the model without that I would be happy to look at a PR, but can you explain a use case?

**Comment by jlmelville** [comment]
Closed by #64
